models/darknet/darknet.hpp

```diff
@@ -0,0 +1,311 @@
+/**
+ * @file darknet.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Darknet models.
+ * 
+ * For more information, kindly refer to the following paper.
+ * 
+ * Paper for DarkNet-19.
+ *
+ * @code
+ * @article{Redmon2016,
+ *  author = {Joseph Redmon, Ali Farhadi},
+ *  title = {YOLO9000 : Better, Faster, Stronger},
+ *  year = {2016},
+ *  url = {https://pjreddie.com/media/files/papers/YOLO9000.pdf}
+ * }
+ * @endcode
+ * 
+ * Paper for DarkNet-53.
+ * 
+ * @code
+ * @article{Redmon2016,
+ *  author = {Joseph Redmon, Ali Farhadi},
+ *  title = {YOLOv3 :  An Incremental Improvement},
+ *  year = {2019},
+ *  url = {https://pjreddie.com/media/files/papers/YOLOv3.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_DARKNET_HPP
+#define MODELS_DARKNET_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/init_rules/he_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */{
+
+/**
+ * Definition of a Darknet CNN.
+ * 
+ * @tparam OutputLayerType The output layer type used to evaluate the network.
+ * @tparam InitializationRuleType Rule used to initialize the weight matrix.
+ * @tparam DaknetVer Version of DarkNet.
+ */
+template<
+  typename OutputLayerType = NegativeLogLikelihood<>,
+  typename InitializationRuleType = HeInitialization,
+  size_t DarkNetVer = 19
```

Comment:
Do you think we can use `DarkNetVersion` here just to make it super clear?

Reply:
Sure.